labonneboite/tests/web/api/test_api.py

```diff
@@ -653,6 +654,30 @@ def test_contact_mode(self):
         self.assertEqual(data['companies'][0]['siret'], u'00000000000004')
         self.assertEqual(data['companies'][0]['contact_mode'], u'Envoyer un CV et une lettre de motivation')
 
+    def test_flag_alternance_false(self):
+        params = self.add_security_params({
+            'commune_id': self.positions['caen']['commune_id'],
+            'rome_codes': u'D1405',
+            'user': u'labonneboite',
+        })
+        rv = self.app.get('/api/v1/company/?%s' % urlencode(params))
```

Comment:
On peut utiliser `url_for` ici à la place d'une url en dur ?

Reply:
Cela me donne une RuntimeException : `Attempted to generate a URL without the application context being pushed. This has to be executed when application context is available.`

Reply:
Hé oui ! Il faut utiliser le `test_request_context`. Regarde dans les autres tests comment il est utilisé.
C'est important d'utiliser `url_for` au cas où les urls changent (c'est du DNRY).
Et c'est important d'utiliser un `test_request_context` pour que la requête en cours soit utilisable par tous les objets dans le cadre du test.

Reply:
Je l'ai rajouté pour chaque test (même si je me suis demandé s'il existait une méthode plus élégante pour ça... !)